src/include/txn.i

```diff
@@ -887,7 +887,7 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
      * Return null not tombstone if nothing is found in history store.
      */
     WT_ASSERT(
-      session, upd == NULL || upd->type != WT_UPDATE_BIRTHMARK || upd->type != WT_UPDATE_TOMBSTONE);
+      session, upd == NULL || (upd->type != WT_UPDATE_BIRTHMARK && upd->type != WT_UPDATE_TOMBSTONE));
```

Comment:
Was getting complier errors for redundant assert. I think this modification looks more correct. Is this ok?